### PR TITLE
FYST-787 Idaho Screeners

### PR DESCRIPTION
**Pull request by DrewProebstel**
## Link to pivotal/JIRA issue
https://codeforamerica.atlassian.net/browse/FYST-787
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- Screener Question were added to the Idaho Flow of FYST
- eligibility_lived_in_state and eligibility_out_of_state_income were pulled from the state_file_base_intake as they did not belong on the Idaho State File Intake.
## How to test?
- Describe the testing approach taken to verify the changes, including:
  - Tests were written for the new controller and new form
  - Idaho FYST feature test was updated with new screener content
- Risk Assessment
  - Opposite of DRY on some of state file intakes, could increase complexity.
## Screenshots (for visual changes)
- After
image

### CODE REVIEW

**Review by powersurge360 — COMMENTED** [review]
Seems fine to me. Merge main and get to green and I'll approve.

**Review by powersurge360 — APPROVED** [review]
Seems good to me.

File: app/lib/navigation/state_file_id_question_navigation.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class StateFileIdQuestionNavigation < Navigation::StateFileBaseQuestionNavigatio
 
     SECTIONS = [
       Navigation::NavigationSection.new("state_file.navigation.section_1", [
-        # eligibility
+        Navigation::NavigationStep.new(StateFile::Questions::IdEligibilityResidenceController)
```
Comment (embarnard) [inline app/lib/navigation/state_file_id_question_navigation.rb:7]: I think we'll probably have to include these two pages in the navigation as well:
```
Navigation::NavigationStep.new(StateFile::Questions::EligibilityOffboardingController, false),
    Navigation::NavigationStep.new(StateFile::Questions::EligibleController),
```

File: spec/features/state_file/complete_intake_spec.rb [hunk 2]
```diff
@@ -430,6 +430,14 @@
...
       click_on I18n.t('general.get_started'), id: "firstCta"
 
+      filing_year = Rails.configuration.statefile_current_tax_year
+      expect(page).to have_text I18n.t("state_file.questions.id_eligibility_residence.edit.emergency_rental_assistance")
+      expect(page).to have_text I18n.t("state_file.questions.id_eligibility_residence.edit.withdrew_msa_fthb", filing_year: filing_year)
+
+      find('#state_file_id_eligibility_residence_form_eligibility_withdrew_msa_fthb_no').click
+      find('#state_file_id_eligibility_residence_form_eligibility_emergency_rental_assistance_no').click
+      click_on I18n.t("general.continue")
+
```
Comment (embarnard) [inline spec/features/state_file/complete_intake_spec.rb:440]: should hit eligible controller before going on to other steps as well

**Pull request merged**